src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs

```diff
@@ -59,6 +67,26 @@ protected static SymbolDisplayPart NewLine()
                 parameter.ToMinimalDisplayParts(semanticModel, position, s_allowDefaultLiteralFormat));
         }
 
+        /// <summary>
+        /// Return the items to display.
+        /// If we have selectedSymbol and selectedParameterIndex, we will highlight the correct parameter.
+        /// Otherwise, we try to guess which parameter to highlight based on the cursor's position in the arguments.
+        /// </summary>
+        protected static SignatureHelpItems MakeSignatureHelpItems(IList<SignatureHelpItem> items, Text.TextSpan textSpan,
+            IMethodSymbol? selectedSymbol, int selectedParameterIndex, int? selectedItem,
+            SeparatedSyntaxList<ArgumentSyntax> arguments, int position)
+        {
+            if (selectedSymbol is null || selectedParameterIndex < 0)
+            {
+                var argumentIndex = LightweightOverloadResolution.TryGetArgumentIndex(arguments, position);
+                return new SignatureHelpItems(items, textSpan, argumentIndex < 0 ? 0 : argumentIndex, arguments.Count, argumentName: null, selectedItem);
```

Comment:
should TryGet just return 0 on failure so you don't need this comparison logic?

Reply:
`TryGetArgumentIndex` is also used elsewhere